### PR TITLE
Feature/powershell script migration

### PR DESCRIPTION
**Pull request by fpodshivadlov**
- Migrated PS script to nodeJS side
- Updated JavaScript formatting
- Implemented force approach for separate files
- Fixed csproj formatting to avoid useless changes in source control
- Added project guid generation to avoid useless changes in source control
- Removed useless files from project

### CODE REVIEW

File: lib/base-generator.js [hunk 1]
```diff
@@ -1,58 +1,70 @@
...
+    return diskFiles.map(sourceFilePath => {
+      const commonPath = utils.getCommonPath(sourcePath);
+      const toFile = path.relative(commonPath, sourceFilePath);
+      let destPath = path.join(destinationPath, toFile);
+      destPath = !customOptions.preProcessPath ? destPath : customOptions.preProcessPath(destPath);
+      return { sourceFilePath, destPath };
+    });
+  }
+
+  _updateFileContent(filePath, pipeline) {
```
Comment (ashishagin) [inline lib/base-generator.js]: Better use transformations, pipeline is too abstract for this specific case

File: lib/extensions.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+const path = require('path');
+
+module.exports = {
+
+  withConflictExclusions: function(BaseGenerator) {
+    return class extends BaseGenerator {
+      constructor(args, opts) {
+        super(args, opts);
+
+        this._conflictExclusions = [];
```
Comment (ashishagin) [inline lib/extensions.js]: Please use map rather than array, that will prevent duplicates and will be more efficient

File: lib/extensions.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        this._extendConflicter();
+      }
+
+      _updateFileContent(filePath, pipeline, options) {
+        const { force } = options || {};
```
Comment (ashishagin) [inline lib/extensions.js]: Please const { force } = options || { force : false}; for better code safety and less further double checks

File: lib/extensions.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+        super._updateFileContent(filePath, pipeline);
+
+        if (force === true) {
+          this._conflictExclusions.push(filePath);
+        }
+      }
+
+      _extendConflicter() {
```
Comment (ashishagin) [inline lib/extensions.js]: That looks like direct patching, let's try to make it more presentable, smth like

```

const Conflicter = require('yeoman-generator/lib/util/conflicter');

class EnhancedConflicter extends Conflicter {

    constructor(args, opts) {
        super(args, opts);
        
        this.forceMap = {};
    }

    collision(file, cb) {
        if (!forceMap.contains(file)) {
            super.collision(file, cb);
            return;
        }

        this.adapter.log.force(path.relative(process.cwd(), file.path));
        cb('force');
    }
}

module.exports =  EnhancedConflicter;

```

Comment (ashishagin) [reply lib/extensions.js]: And then set enhanced conflicter in consturctor

File: lib/extensions.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+const path = require('path');
+
+module.exports = {
+
+  withConflictExclusions: function(BaseGenerator) {
```
Comment (ashishagin) [inline lib/extensions.js]: We can call it enhanced conflicted and by that cover all possible future changes \ additions

File: lib/base-generator.js [hunk 6]
```diff
@@ -1,58 +1,70 @@
...
+  _baseGlobOptions() {
+    return {
+      dot: true,
+      sync: true,
+      debug: false,
+    };
+  }
 }
+
+module.exports = extensions.withConflictExclusions(BaseGenerator);
```
Comment (ashishagin) [inline lib/base-generator.js]: Please use it as decorator

File: lib/extensions.js (hunk 4 above)
Comment (fpodshivadlov) [reply lib/extensions.js]: The original `Conflicter` class is not exported from module - I didn't find better solution rather that use this way.

File: generators/module/index.js [hunk 7]
```diff
@@ -88,79 +89,79 @@ module.exports = class extends BaseGenerator {
...
         name: 'sitecoreUpdate',
         message: msg.sitecoreUpdate.prompt,
         choices: this.options.sitecoreVersion.value && this.options.sitecoreVersion.value,
         when: !this.options.sitecoreVersion,
       },
     ]);
+
+    this.options = { ...this.options, ...answers };
+
+    this.options.codeGuidSeed = `${this.options.solutionName}.${this.options.moduleType}.${this.options.moduleName}`;
```
[the remark this answers is not in the document]
Comment (fpodshivadlov) [reply generators/module/index.js]: It's used in `_copyAll` as well

File: lib/base-generator.js (hunk 6 above)
Comment (fpodshivadlov) [reply lib/base-generator.js]: It is not supported by Node at the moment

File: lib/extensions.js (hunk 5 above)
Comment (ashishagin) [reply lib/extensions.js]: Please rename it to with enhancedConflicted

**Pull request merged**